src/client/mod.rs

```diff
@@ -157,41 +157,51 @@ impl APIClient {
         trace!("{} {}", res.status().as_str(), res.url());
 
         // Now use `unfold` to convert the chunked responses into a Stream
-        Ok(futures::stream::unfold(res, |mut resp| {
-            async move {
-                // A buffer to store our current object in
-                // as we stream it's entirety
-                let mut buff = Vec::new();
-
-                loop {
-                    match resp.chunk().await {
-                        Ok(Some(chunk)) => {
-                            // append it to our current buffer
-                            buff.extend_from_slice(&chunk);
-
-                            // Check to see if we have valid JSON in our buffer,
-                            // and if so parse it fully and return.
-                            // Deserializing as serde::de::IgnoredAny is just a syntax check
-                            // and does not go through any of the memory intensity of constructing the actual
-                            // value.
-                            let valid = serde_json::from_slice::<IgnoredAny>(&buff);
-
-                            if valid.is_ok() {
-                                match serde_json::from_slice::<T>(&buff) {
-                                    Ok(val) => return Some((Ok(val), resp)),
-                                    Err(e) => {
-                                        warn!("{} {:?}", String::from_utf8_lossy(&buff), e);
-                                        return Some((Err(Error::SerdeError(e)), resp));
+        Ok(futures::stream::unfold(
+            (res, Vec::new(), Vec::new()),
+            |(mut resp, mut buff, mut items): (_, _, Vec<Result<T>>)| {
+                async {
+                    // If we have any items, pop off the first,
+                    // yield it, and then continue into the next iteration
+                    if !items.is_empty() {
+                        let current = items.pop().unwrap(); // We know items is not empty so this is safe
+                        return Some((current, (resp, buff, items)));
+                    }
```

Comment:
First; this should be an `if let Some(current) = items.first()` to avoid the unwrap.

Second; why is this done at the beginning of the next iteration rather than the end of the previous? It reads a bit strange this way. It also causes us to keep an extra mutable in between the fold (and probably also causes the Send requirement)